core/jazz_es-kinesis-log-streamer/components/utils.js

```diff
@@ -136,96 +200,235 @@ function getApiLogsData(payload) {
-  let bulkRequestBody = '',
-    data = {}
-  data.asset_type = "lambda";
-  data.provider = "aws";
-  let domainAndservice, serviceInfo;
-  data.request_id = getInfo(payload.logEvents, config.PATTERNS.lambda_request_id);
-  if (data.request_id) {
-    data.asset_identifier = payload.logGroup.split(config.PATTERNS.asset_identifier_key)[1];
-    let environment = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment, 2);
-		if (environment === "dev") {
-			let dev_environment = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment_dev, 2);
-			serviceInfo = getSubInfo(payload.logGroup, config.PATTERNS.lambda_environment_dev, 1);
-			data.environment = dev_environment;
-		} else {
-			data.environment = environment;
-			serviceInfo = getSubInfo(payload.logGroup, config.PATTERNS.lambda_domain_service, 1);
-		}
-
-		domainAndservice = serviceInfo;
-		if(serviceInfo.indexOf(config.PATTERNS.sls_app_function) !== -1 ) { // if yes then sls-app function
-			let serviceInfoArr = serviceInfo.split(config.PATTERNS.sls_app_function);
-			domainAndservice = serviceInfoArr[0];
-		} 
-
-		logger.debug("domainAndservice: " + domainAndservice)
-    let domain = domainAndservice.substring(0, domainAndservice.indexOf("_"));
-    if (domain) {
-      data.domain = domain;
-      data.servicename = domainAndservice.substring(domain.length + 1);
-    } else {
-      data.domain = "";
-      data.servicename = domainAndservice;
-    }
-
-    if (data.servicename) {
-      payload.logEvents.forEach(function (logEvent) {
-
-        data.request_id = getSubInfo(logEvent.message, config.PATTERNS.guid_regex, 0);
+function getLambdaLogsData(configValue, payload, callback) {
+  // first get token for calling respective APIs
+  getToken(configValue)
+  .then((creds) => {
+    // get configDB data for getting roleArn specific to account
+    getConfigJson(configValue, creds)
+    .then((configData) => {
+      // get accountId and region through service Data
+      getsServiceMetaData(configValue, payload.logGroup, creds)
+      .then((serviceData) => {
+        // execute sts:assumeRole
+        assumeRole(configData, serviceData)
+        .then((tempCreds) => {
+          // configure cloudwatch and retrieve tags from logGroup
+          getLogsGroupsTags(payload.logGroup, tempCreds, serviceData)
+          .then((tagsResult) => {
+            let bulkRequestBody = '',
+            data = {}
+            data.asset_type = "lambda";
+            data.provider = "aws";
+            let domainAndservice, serviceInfo, environment, dev_environment, serviceInfoArr, domain;
+            data.request_id = getInfo(payload.logEvents, config.PATTERNS.lambda_request_id);
+            if (data.request_id) {
+              data.asset_identifier = payload.logGroup.split(config.PATTERNS.asset_identifier_key)[1];
+              // if tags present, then it is for sls-app, otherwise for other services
+              if(tagsResult != 'error' && tagsResult.tags.environment){
```

Comment:
Add check for all the tags (not just `tagsResult.tags.environment`).

Reply:
Done.